resdk/resources/base.py

```diff
@@ -28,37 +28,40 @@ class BaseResource(object):
 
     def __init__(self, slug=None, id=None,  # pylint: disable=redefined-builtin
                  model_data=None, resolwe=None):
-        """Verify that one and only one of slug, id, model_data is given."""
-        identifiers = iter((slug, id, model_data))
-        if not any(identifiers) or any(identifiers):
-            raise ValueError("One and only one of slug, id or model_data must be given")
+        """Verify that only a single attribute of slug, id or model_data given."""
+        if len([attr for attr in (slug, id, model_data) if attr is not None]) > 1:
+            raise ValueError("Only one of slug, id or model_data allowed")
 
         if id:
             if not isinstance(id, int):
                 raise ValueError("id should be an integer.")
         elif slug:
             if not isinstance(slug, six.string_types):
                 raise ValueError("slug should be a string.")
-        else:
+        elif model_data:
             if not isinstance(model_data, dict):
                 raise ValueError("model_data should be a dict.")
 
+        self._add_fields('_writable_fields', ('slug', 'name', 'permissions'))
```

Comment:
Maybe we should define this (_writable_fields, _update_protected_fields, _read_only_fields) as class attributes. The class instance should not change this lists of fields.